### PR TITLE
split flag section into common and advanced flags

### PR DESCRIPTION
**Pull request by RalfJung**
As discussed with @oli-obk . However I was not always sure which flags to put where, so if you think some flags should be in the other category please let me know. :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -305,6 +284,44 @@ to Miri failing to detect cases of undefined behavior in a program.
   `-Zmiri-disable-validation` is set.
```
Comment (oli-obk) [inline README.md:284]: Huh, why is env var isolation affected by validation?

File: README.md [hunk 2]
```diff
@@ -305,6 +284,44 @@ to Miri failing to detect cases of undefined behavior in a program.
...
   remaining threads to exist when the main thread exits.
+* `-Zmiri-seed=<hex>` configures the seed of the RNG that Miri uses to resolve
+  non-determinism. This RNG is used to pick base addresses for allocations. When
+  isolation is enabled (the default), this is also used to emulate system
+  entropy. The default seed is 0. You can increase test coverage by running Miri
+  multiple times with different seeds.
+  **NOTE**: This entropy is not good enough for cryptographic use! Do not
+  generate secret keys in Miri or perform other kinds of cryptographic
+  operations that rely on proper random numbers.
+* `-Zmiri-strict-provenance` enables [strict
```
Comment (oli-obk) [inline README.md:295]: Is this common at present, seems experimental. Once it leaves experimental mode, we should probably switch to it by default and add a disable flag. Though seems ok here for visibility

**Review by oli-obk — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (RalfJung) [reply README.md:284]: Oh lol that should be talking about isolation, not validation.^^

File: README.md (hunk 2 above)
Comment (RalfJung) [reply README.md:295]: I don't think strict-provenance should be the default, see https://github.com/rust-lang/miri/issues/2133.

**Pull request merged**